### ISSUE TITLE
[Investigation] Descender-cut.

### ISSUE THREAD

**Issue by ShikiSuen** [adobe-fonts/source-han-sans#100]
I had ever met this issue in SHS 1.001 after changing LineGap to zero until I use ascender and descender inherited from Hiragino Kaku Gothic's winasc and windesc:
![image](https://cloud.githubusercontent.com/assets/3164826/7235509/d6ee4662-e74b-11e4-9e6a-959b5f3617af.png)

Probably the intel above could be taken as a reference while you are investigating this problem (SHS 1.002, probably not only in LibreOffice):
![image](https://cloud.githubusercontent.com/assets/3164826/7235539/0c4220e0-e74c-11e4-9b42-49ad800969d1.png)
![image](https://cloud.githubusercontent.com/assets/3164826/7235572/246b18fc-e74c-11e4-85c4-2f4aec9b6b6d.png)

**Comment by ShikiSuen** [comment]
One more thing: This issue is probably out of issue #76 since Adobe Official SHS 1.002 doesn't have that issue anymore:
![image](https://cloud.githubusercontent.com/assets/3164826/7237004/ba308b8e-e755-11e4-929d-ddea080be8aa.png)

**Comment by ShikiSuen** [comment]
//I removed my last reply and use this one instead:
One more thing: This issue is probably out of issue #41 since Adobe Official SHS 1.002 doesn't have that issue anymore:

**Comment by ShikiSuen** [comment]
@kenlunde I forked out the Source Han Sans UI 1.002 on my laptop, but I have found that Descender -234 might not be a good choice for SHS 1.002. The following screenshot still shows a little-bit descender-cut:
![image](https://cloud.githubusercontent.com/assets/3164826/7259427/7622654a-e820-11e4-8709-92b6c5eb9553.png)

Could you please provide some hints regarding whether I should turn the absolute value of descender larger than 234 or not? Thanks.

**Comment by ShikiSuen** [comment]
//I removed my last reply and use this one instead:
@kenlunde I forked out the Source Han Sans UI 1.002 on my laptop, but I have found that Descender -234 might not be a good choice for SHS 1.002. The following screenshot still shows a little-bit descender-cut (regarding selection only):
![image](https://cloud.githubusercontent.com/assets/3164826/7259484/c8e779f0-e820-11e4-927f-fd9e739ae7ab.png)

Could you please provide some hints regarding whether I should turn the absolute value of descender larger than 234 or not? Thanks.

**Comment by ShikiSuen** [comment]
I removed my previous reply 'cause those ascender and descender parameters I provided above could let SHS 1.002 behaves as same as SHS-UI 1.001, looks like I could left this thread alone at this moment.
(Probably I shouldn't only use LibreOffice to test these things.)

**Comment by kenlunde** [comment]
LibreOffice indeed seems like a less-than-ideal platform for testing or checking various vertical metrics settings. The testing that Adobe and Google performed indicated that the Version 1.002 settings are better than those in the previous versions. Feel free to close this issue if you feel doing so is appropriate.

**Comment by ShikiSuen** [comment]
OK then I close this one first, you may want to reopen this one if you felt that the following one could be treated as a problem (displayed in OS X Fontbook, but I think it might be CoreText's problem):
![image](https://cloud.githubusercontent.com/assets/3164826/7265776/6acd12c6-e85c-11e4-8253-7d68247516a7.png)

**Comment by kenlunde** [comment]
For now, let's leave this as closed/to-track.

**Comment by ShikiSuen** [comment]
@kenlunde Do you have any recommendations regarding tools of font metric testing?

**Comment by kenlunde** [comment]
Any modern browser application.

**Comment by ShikiSuen** [comment]
^ Got it, I will check it out.

One more thing: I have found that the file size of SuperOTC could be obviously larger since I modified Ascender and Descender to the one I posted above (no other parameters are added among features.otc files), the file sizes are shown in the following pic:
![image](https://cloud.githubusercontent.com/assets/3164826/7265859/e117f26a-e85d-11e4-969f-82e9c03e92a8.png)

Till now, I couldn't figure out why the file size increases so far. This made SHS-UI not upload-able to GitHub after zip compression.

P.S.: The script for building SHS-UI 1.002 are the following:
<pre><code>cd /Users/ShikiSuen/Desktop/FontWorkingFolder/source-han-sans-1.002/Bold/OTC
makeotf -f cidfont.ps.OTC.SC -ff features.OTC.SC -fi cidfontinfo.OTC.SC -mf ../../FontMenuNameDB -r -nS -cs 25 -ch ../../UniSourceHanSansCN-UTF32-H
makeotf -f cidfont.ps.OTC.TC -ff features.OTC.TC -fi cidfontinfo.OTC.TC -mf ../../FontMenuNameDB -r -nS -cs 2 -ch ../../UniSourceHanSansTW-UTF32-H -ci ../../SourceHanSans_TWHK_sequences.txt
makeotf -f cidfont.ps.OTC.J -ff features.OTC.J -fi cidfontinfo.OTC.J -mf ../../FontMenuNameDB -r -nS -cs 1 -ch ../../UniSourceHanSansJP-UTF32-H -ci ../../SourceHanSans_JP_sequences.txt
makeotf -f cidfont.ps.OTC.K -ff features.OTC.K -fi cidfontinfo.OTC.K -mf ../../FontMenuNameDB -r -nS -cs 3 -ch ../../UniSourceHanSansKR-UTF32-H -ci ../../SourceHanSans_KR_sequences.txt
makeotf -f cidfont.ps.OTC.HW.SC -ff features.OTC.SC -fi cidfontinfo.OTC.HW.SC -mf ../../FontMenuNameDB.HW -r -nS -cs 25 -ch ../../UniSourceHanSansHWCN-UTF32-H
makeotf -f cidfont.ps.OTC.HW.TC -ff features.OTC.TC -fi cidfontinfo.OTC.HW.TC -mf ../../FontMenuNameDB.HW -r -nS -cs 2 -ch ../../UniSourceHanSansHWTW-UTF32-H -ci ../../SourceHanSans_TWHK_sequences.txt
makeotf -f cidfont.ps.OTC.HW.J -ff features.OTC.J -fi cidfontinfo.OTC.HW.J -mf ../../FontMenuNameDB.HW -r -nS -cs 1 -ch ../../UniSourceHanSansHWJP-UTF32-H -ci ../../SourceHanSans_JP_sequences.txt
makeotf -f cidfont.ps.OTC.HW.K -ff features.OTC.K -fi cidfontinfo.OTC.HW.K -mf ../../FontMenuNameDB.HW -r -nS -cs 3 -ch ../../UniSourceHanSansHWKR-UTF32-H -ci ../../SourceHanSans_KR_sequences.txt

cd /Users/ShikiSuen/Desktop/FontWorkingFolder/source-han-sans-1.002/ExtraLight/OTC
makeotf -f cidfont.ps.OTC.SC -ff features.OTC.SC -fi cidfontinfo.OTC.SC -mf ../../FontMenuNameDB -r -nS -cs 25 -ch ../../UniSourceHanSansCN-UTF32-H
makeotf -f cidfont.ps.OTC.TC -ff features.OTC.TC -fi cidfontinfo.OTC.TC -mf ../../FontMenuNameDB -r -nS -cs 2 -ch ../../UniSourceHanSansTW-UTF32-H -ci ../../SourceHanSans_TWHK_sequences.txt
makeotf -f cidfont.ps.OTC.J -ff features.OTC.J -fi cidfontinfo.OTC.J -mf ../../FontMenuNameDB -r -nS -cs 1 -ch ../../UniSourceHanSansJP-UTF32-H -ci ../../SourceHanSans_JP_sequences.txt
makeotf -f cidfont.ps.OTC.K -ff features.OTC.K -fi cidfontinfo.OTC.K -mf ../../FontMenuNameDB -r -nS -cs 3 -ch ../../UniSourceHanSansKR-UTF32-H -ci ../../SourceHanSans_KR_sequences.txt

cd /Users/ShikiSuen/Desktop/FontWorkingFolder/source-han-sans-1.002/Heavy/OTC
makeotf -f cidfont.ps.OTC.SC -ff features.OTC.SC -fi cidfontinfo.OTC.SC -mf ../../FontMenuNameDB -r -nS -cs 25 -ch ../../UniSourceHanSansCN-UTF32-H
makeotf -f cidfont.ps.OTC.TC -ff features.OTC.TC -fi cidfontinfo.OTC.TC -mf ../../FontMenuNameDB -r -nS -cs 2 -ch ../../UniSourceHanSansTW-UTF32-H -ci ../../SourceHanSans_TWHK_sequences.txt
makeotf -f cidfont.ps.OTC.J -ff features.OTC.J -fi cidfontinfo.OTC.J -mf ../../FontMenuNameDB -r -nS -cs 1 -ch ../../UniSourceHanSansJP-UTF32-H -ci ../../SourceHanSans_JP_sequences.txt
makeotf -f cidfont.ps.OTC.K -ff features.OTC.K -fi cidfontinfo.OTC.K -mf ../../FontMenuNameDB -r -nS -cs 3 -ch ../../UniSourceHanSansKR-UTF32-H -ci ../../SourceHanSans_KR_sequences.txt

cd /Users/ShikiSuen/Desktop/FontWorkingFolder/source-han-sans-1.002/Light/OTC
makeotf -f cidfont.ps.OTC.SC -ff features.OTC.SC -fi cidfontinfo.OTC.SC -mf ../../FontMenuNameDB -r -nS -cs 25 -ch ../../UniSourceHanSansCN-UTF32-H
makeotf -f cidfont.ps.OTC.TC -ff features.OTC.TC -fi cidfontinfo.OTC.TC -mf ../../FontMenuNameDB -r -nS -cs 2 -ch ../../UniSourceHanSansTW-UTF32-H -ci ../../SourceHanSans_TWHK_sequences.txt
makeotf -f cidfont.ps.OTC.J -ff features.OTC.J -fi cidfontinfo.OTC.J -mf ../../FontMenuNameDB -r -nS -cs 1 -ch ../../UniSourceHanSansJP-UTF32-H -ci ../../SourceHanSans_JP_sequences.txt
makeotf -f cidfont.ps.OTC.K -ff features.OTC.K -fi cidfontinfo.OTC.K -mf ../../FontMenuNameDB -r -nS -cs 3 -ch ../../UniSourceHanSansKR-UTF32-H -ci ../../SourceHanSans_KR_sequences.txt

cd /Users/ShikiSuen/Desktop/FontWorkingFolder/source-han-sans-1.002/Medium/OTC
makeotf -f cidfont.ps.OTC.SC -ff features.OTC.SC -fi cidfontinfo.OTC.SC -mf ../../FontMenuNameDB -r -nS -cs 25 -ch ../../UniSourceHanSansCN-UTF32-H
makeotf -f cidfont.ps.OTC.TC -ff features.OTC.TC -fi cidfontinfo.OTC.TC -mf ../../FontMenuNameDB -r -nS -cs 2 -ch ../../UniSourceHanSansTW-UTF32-H -ci ../../SourceHanSans_TWHK_sequences.txt
makeotf -f cidfont.ps.OTC.J -ff features.OTC.J -fi cidfontinfo.OTC.J -mf ../../FontMenuNameDB -r -nS -cs 1 -ch ../../UniSourceHanSansJP-UTF32-H -ci ../../SourceHanSans_JP_sequences.txt
makeotf -f cidfont.ps.OTC.K -ff features.OTC.K -fi cidfontinfo.OTC.K -mf ../../FontMenuNameDB -r -nS -cs 3 -ch ../../UniSourceHanSansKR-UTF32-H -ci ../../SourceHanSans_KR_sequences.txt

cd /Users/ShikiSuen/Desktop/FontWorkingFolder/source-han-sans-1.002/Normal/OTC
makeotf -f cidfont.ps.OTC.SC -ff features.OTC.SC -fi cidfontinfo.OTC.SC -mf ../../FontMenuNameDB -r -nS -cs 25 -ch ../../UniSourceHanSansCN-UTF32-H
makeotf -f cidfont.ps.OTC.TC -ff features.OTC.TC -fi cidfontinfo.OTC.TC -mf ../../FontMenuNameDB -r -nS -cs 2 -ch ../../UniSourceHanSansTW-UTF32-H -ci ../../SourceHanSans_TWHK_sequences.txt
makeotf -f cidfont.ps.OTC.J -ff features.OTC.J -fi cidfontinfo.OTC.J -mf ../../FontMenuNameDB -r -nS -cs 1 -ch ../../UniSourceHanSansJP-UTF32-H -ci ../../SourceHanSans_JP_sequences.txt
makeotf -f cidfont.ps.OTC.K -ff features.OTC.K -fi cidfontinfo.OTC.K -mf ../../FontMenuNameDB -r -nS -cs 3 -ch ../../UniSourceHanSansKR-UTF32-H -ci ../../SourceHanSans_KR_sequences.txt

cd /Users/ShikiSuen/Desktop/FontWorkingFolder/source-han-sans-1.002/Regular/OTC
makeotf -f cidfont.ps.OTC.SC -ff features.OTC.SC -fi cidfontinfo.OTC.SC -mf ../../FontMenuNameDB -r -nS -cs 25 -ch ../../UniSourceHanSansCN-UTF32-H
makeotf -f cidfont.ps.OTC.TC -ff features.OTC.TC -fi cidfontinfo.OTC.TC -mf ../../FontMenuNameDB -r -nS -cs 2 -ch ../../UniSourceHanSansTW-UTF32-H -ci ../../SourceHanSans_TWHK_sequences.txt
makeotf -f cidfont.ps.OTC.J -ff features.OTC.J -fi cidfontinfo.OTC.J -mf ../../FontMenuNameDB -r -nS -cs 1 -ch ../../UniSourceHanSansJP-UTF32-H -ci ../../SourceHanSans_JP_sequences.txt
makeotf -f cidfont.ps.OTC.K -ff features.OTC.K -fi cidfontinfo.OTC.K -mf ../../FontMenuNameDB -r -nS -cs 3 -ch ../../UniSourceHanSansKR-UTF32-H -ci ../../SourceHanSans_KR_sequences.txt
makeotf -f cidfont.ps.OTC.HW.SC -ff features.OTC.SC -fi cidfontinfo.OTC.HW.SC -mf ../../FontMenuNameDB.HW -r -nS -cs 25 -ch ../../UniSourceHanSansHWCN-UTF32-H
makeotf -f cidfont.ps.OTC.HW.TC -ff features.OTC.TC -fi cidfontinfo.OTC.HW.TC -mf ../../FontMenuNameDB.HW -r -nS -cs 2 -ch ../../UniSourceHanSansHWTW-UTF32-H -ci ../../SourceHanSans_TWHK_sequences.txt
makeotf -f cidfont.ps.OTC.HW.J -ff features.OTC.J -fi cidfontinfo.OTC.HW.J -mf ../../FontMenuNameDB.HW -r -nS -cs 1 -ch ../../UniSourceHanSansHWJP-UTF32-H -ci ../../SourceHanSans_JP_sequences.txt
makeotf -f cidfont.ps.OTC.HW.K -ff features.OTC.K -fi cidfontinfo.OTC.HW.K -mf ../../FontMenuNameDB.HW -r -nS -cs 3 -ch ../../UniSourceHanSansHWKR-UTF32-H -ci ../../SourceHanSans_KR_sequences.txt

cd /Users/ShikiSuen/Desktop/FontWorkingFolder/source-han-sans-1.002/Bold/OTC
sfntedit -x CFF=CFF.J Adobe-SHSDesk-Bold.otf
sfntedit -a CFF=CFF.J Adobe-SHSDeskK-Bold.otf
sfntedit -a CFF=CFF.J Adobe-SHSDeskSC-Bold.otf
sfntedit -a CFF=CFF.J Adobe-SHSDeskTC-Bold.otf
sfntedit -x CFF=CFF.J Adobe-SHSDeskHW-Bold.otf
sfntedit -a CFF=CFF.J Adobe-SHSDeskHWK-Bold.otf
sfntedit -a CFF=CFF.J Adobe-SHSDeskHWSC-Bold.otf
sfntedit -a CFF=CFF.J Adobe-SHSDeskHWTC-Bold.otf
sfntedit -d DSIG Adobe-SHSDesk-Bold.otf
sfntedit -d DSIG Adobe-SHSDeskK-Bold.otf
sfntedit -d DSIG Adobe-SHSDeskSC-Bold.otf
sfntedit -d DSIG Adobe-SHSDeskTC-Bold.otf
sfntedit -d DSIG Adobe-SHSDeskHW-Bold.otf
sfntedit -d DSIG Adobe-SHSDeskHWK-Bold.otf
sfntedit -d DSIG Adobe-SHSDeskHWSC-Bold.otf
sfntedit -d DSIG Adobe-SHSDeskHWTC-Bold.otf

cd /Users/ShikiSuen/Desktop/FontWorkingFolder/source-han-sans-1.002/ExtraLight/OTC
sfntedit -x CFF=CFF.J Adobe-SHSDesk-ExtraLight.otf
sfntedit -a CFF=CFF.J Adobe-SHSDeskK-ExtraLight.otf
sfntedit -a CFF=CFF.J Adobe-SHSDeskSC-ExtraLight.otf
sfntedit -a CFF=CFF.J Adobe-SHSDeskTC-ExtraLight.otf
sfntedit -d DSIG Adobe-SHSDesk-ExtraLight.otf
sfntedit -d DSIG Adobe-SHSDeskK-ExtraLight.otf
sfntedit -d DSIG Adobe-SHSDeskSC-ExtraLight.otf
sfntedit -d DSIG Adobe-SHSDeskTC-ExtraLight.otf

cd /Users/ShikiSuen/Desktop/FontWorkingFolder/source-han-sans-1.002/Heavy/OTC
sfntedit -x CFF=CFF.J Adobe-SHSDesk-Heavy.otf
sfntedit -a CFF=CFF.J Adobe-SHSDeskK-Heavy.otf
sfntedit -a CFF=CFF.J Adobe-SHSDeskSC-Heavy.otf
sfntedit -a CFF=CFF.J Adobe-SHSDeskTC-Heavy.otf
sfntedit -d DSIG Adobe-SHSDesk-Heavy.otf
sfntedit -d DSIG Adobe-SHSDeskK-Heavy.otf
sfntedit -d DSIG Adobe-SHSDeskSC-Heavy.otf
sfntedit -d DSIG Adobe-SHSDeskTC-Heavy.otf

cd /Users/ShikiSuen/Desktop/FontWorkingFolder/source-han-sans-1.002/Light/OTC
sfntedit -x CFF=CFF.J Adobe-SHSDesk-Light.otf
sfntedit -a CFF=CFF.J Adobe-SHSDeskK-Light.otf
sfntedit -a CFF=CFF.J Adobe-SHSDeskSC-Light.otf
sfntedit -a CFF=CFF.J Adobe-SHSDeskTC-Light.otf
sfntedit -d DSIG Adobe-SHSDesk-Light.otf
sfntedit -d DSIG Adobe-SHSDeskK-Light.otf
sfntedit -d DSIG Adobe-SHSDeskSC-Light.otf
sfntedit -d DSIG Adobe-SHSDeskTC-Light.otf

cd /Users/ShikiSuen/Desktop/FontWorkingFolder/source-han-sans-1.002/Medium/OTC
sfntedit -x CFF=CFF.J Adobe-SHSDesk-Medium.otf
sfntedit -a CFF=CFF.J Adobe-SHSDeskK-Medium.otf
sfntedit -a CFF=CFF.J Adobe-SHSDeskSC-Medium.otf
sfntedit -a CFF=CFF.J Adobe-SHSDeskTC-Medium.otf
sfntedit -d DSIG Adobe-SHSDesk-Medium.otf
sfntedit -d DSIG Adobe-SHSDeskK-Medium.otf
sfntedit -d DSIG Adobe-SHSDeskSC-Medium.otf
sfntedit -d DSIG Adobe-SHSDeskTC-Medium.otf

cd /Users/ShikiSuen/Desktop/FontWorkingFolder/source-han-sans-1.002/Normal/OTC
sfntedit -x CFF=CFF.J Adobe-SHSDesk-Normal.otf
sfntedit -a CFF=CFF.J Adobe-SHSDeskK-Normal.otf
sfntedit -a CFF=CFF.J Adobe-SHSDeskSC-Normal.otf
sfntedit -a CFF=CFF.J Adobe-SHSDeskTC-Normal.otf
sfntedit -d DSIG Adobe-SHSDesk-Normal.otf
sfntedit -d DSIG Adobe-SHSDeskK-Normal.otf
sfntedit -d DSIG Adobe-SHSDeskSC-Normal.otf
sfntedit -d DSIG Adobe-SHSDeskTC-Normal.otf

cd /Users/ShikiSuen/Desktop/FontWorkingFolder/source-han-sans-1.002/Regular/OTC
sfntedit -x CFF=CFF.J Adobe-SHSDesk-Regular.otf
sfntedit -a CFF=CFF.J Adobe-SHSDeskK-Regular.otf
sfntedit -a CFF=CFF.J Adobe-SHSDeskSC-Regular.otf
sfntedit -a CFF=CFF.J Adobe-SHSDeskTC-Regular.otf
sfntedit -x CFF=CFF.J Adobe-SHSDeskHW-Regular.otf
sfntedit -a CFF=CFF.J Adobe-SHSDeskHWK-Regular.otf
sfntedit -a CFF=CFF.J Adobe-SHSDeskHWSC-Regular.otf
sfntedit -a CFF=CFF.J Adobe-SHSDeskHWTC-Regular.otf
sfntedit -d DSIG Adobe-SHSDesk-Regular.otf
sfntedit -d DSIG Adobe-SHSDeskK-Regular.otf
sfntedit -d DSIG Adobe-SHSDeskSC-Regular.otf
sfntedit -d DSIG Adobe-SHSDeskTC-Regular.otf
sfntedit -d DSIG Adobe-SHSDeskHW-Regular.otf
sfntedit -d DSIG Adobe-SHSDeskHWK-Regular.otf
sfntedit -d DSIG Adobe-SHSDeskHWSC-Regular.otf
sfntedit -d DSIG Adobe-SHSDeskHWTC-Regular.otf

cd /Users/ShikiSuen/Desktop/FontWorkingFolder/source-han-sans-1.002/
otf2otc -o SourceHanSansUI.ttc ExtraLight/OTC/Adobe-SHSDesk-ExtraLight.otf ExtraLight/OTC/Adobe-SHSDeskK-ExtraLight.otf ExtraLight/OTC/Adobe-SHSDeskSC-ExtraLight.otf ExtraLight/OTC/Adobe-SHSDeskTC-ExtraLight.otf Light/OTC/Adobe-SHSDesk-Light.otf Light/OTC/Adobe-SHSDeskK-Light.otf Light/OTC/Adobe-SHSDeskSC-Light.otf Light/OTC/Adobe-SHSDeskTC-Light.otf Normal/OTC/Adobe-SHSDesk-Normal.otf Normal/OTC/Adobe-SHSDeskK-Normal.otf Normal/OTC/Adobe-SHSDeskSC-Normal.otf Normal/OTC/Adobe-SHSDeskTC-Normal.otf Regular/OTC/Adobe-SHSDesk-Regular.otf Regular/OTC/Adobe-SHSDeskK-Regular.otf Regular/OTC/Adobe-SHSDeskSC-Regular.otf Regular/OTC/Adobe-SHSDeskTC-Regular.otf Regular/OTC/Adobe-SHSDeskHW-Regular.otf Regular/OTC/Adobe-SHSDeskHWK-Regular.otf Regular/OTC/Adobe-SHSDeskHWSC-Regular.otf Regular/OTC/Adobe-SHSDeskHWTC-Regular.otf Medium/OTC/Adobe-SHSDesk-Medium.otf Medium/OTC/Adobe-SHSDeskK-Medium.otf Medium/OTC/Adobe-SHSDeskSC-Medium.otf Medium/OTC/Adobe-SHSDeskTC-Medium.otf Bold/OTC/Adobe-SHSDesk-Bold.otf Bold/OTC/Adobe-SHSDeskK-Bold.otf Bold/OTC/Adobe-SHSDeskSC-Bold.otf Bold/OTC/Adobe-SHSDeskTC-Bold.otf Bold/OTC/Adobe-SHSDeskHW-Bold.otf Bold/OTC/Adobe-SHSDeskHWK-Bold.otf Bold/OTC/Adobe-SHSDeskHWSC-Bold.otf Bold/OTC/Adobe-SHSDeskHWTC-Bold.otf Heavy/OTC/Adobe-SHSDesk-Heavy.otf Heavy/OTC/Adobe-SHSDeskK-Heavy.otf Heavy/OTC/Adobe-SHSDeskSC-Heavy.otf Heavy/OTC/Adobe-SHSDeskTC-Heavy.otf</code></pre>

**Comment by kenlunde** [comment]
You're simply building unsubroutinized CFFs.

You need to build the "master" OTFs (the ones from which you extract the 'CFF ' table and splice it into the other OTFs of the same weight) using the "-maxs 3000" command-line option in lieu of the "-nS" one. Note that this process may take several hours per font. In yours first set of command lines, the fourth one should be as follows:
~~~
makeotf -f cidfont.ps.OTC.J -ff features.OTC.J -fi cidfontinfo.OTC.J -mf ../../FontMenuNameDB -r -maxs 30000 -cs 1 -ch ../../UniSourceHanSansJP-UTF32-H -ci ../../SourceHanSans_JP_sequences.txt
~~~

**Comment by ShikiSuen** [comment]
@kenlunde Thanks for explanation. Due to the heavy computing amount needed regarding such compilation, I will ask someone for help if he's glad, convenient and he has an Urn-like Mac Pro. For this moment, I'd better use official 1.002 instead.

**Comment by ShikiSuen** [comment]
@kenlunde I use "-maxs 3000" instead of "-nS" in all makeotf sentences regarding Japanese Language-Specified OTFs' compilation. But the final SuperOTC file size reduced to 140MB only:
(File Size calculation, since OS X Mountain Lion, is different from the way Windows does.)
![image](https://cloud.githubusercontent.com/assets/3164826/7301663/5b01dd86-e9a1-11e4-8191-41f6b1383975.png)

**Comment by kenlunde** [comment]
Given the difference in size, which is roughly the same as two CFFs, you no doubt forgot to execute the following command lines:

~~~
sfntedit -a CFF=CFF.J SourceHanSansHW-Regular.otf.tmp
sfntedit -a CFF=CFF.J SourceHanSansHW-Bold.otf.tmp
~~~

**Comment by ShikiSuen** [comment]
@kenlunde I modified those two HW CFF commands by changing "-x" into "-a", the final output file size is exactly the same as the official SuperOTF builds. Thanks for your assistance.

**Comment by ShikiSuen** [comment]
This issue has been solved in Source Han Sans 1.003, Please label this issue as solved.

**Comment by kenlunde** [comment]
Fixed in Version 1.003. Thank you, @ShikiSuen, for confirming!

**Comment by ShikiSuen** [comment]
Your welcome.
